pkg/controller/operators/catalog/operator.go

```diff
@@ -1017,17 +1020,36 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 			return err
 		}
 		updatedSubs = o.setIPReference(updatedSubs, maxGeneration+1, installPlanReference)
-		for _, updatedSub := range updatedSubs {
-			for i, sub := range subs {
-				if sub.Name == updatedSub.Name && sub.Namespace == updatedSub.Namespace {
-					subs[i] = updatedSub
-				}
-			}
-		}
 	} else {
 		logger.Debugf("no subscriptions were updated")
 	}
 
+	// Remove resolutionfailed condition from subscriptions
+	subs = o.removeSubsCond(subs, v1alpha1.SubscriptionResolutionFailed)
```

Comment:
Just a quick note: it feels like this function may be a candidate for refactoring in the future. I don't think we need to tackle  that kind of change in this PR but I wanted to get any thoughts on whether you had that same though when poking at this locally?

Reply:
Just copy-pasting the comment chain on the other PR: https://github.com/operator-framework/operator-lifecycle-manager/pull/2429#discussion_r758548598.

Reply:
I'd like to hold off on refactoring and keeping the scope of these changes small as they're affecting upstream/downstream e2e runs, and combing down hot looping behavior in some edge cases.